### DISCUSSION TITLE
Choice by Numbers

### DISCUSSION THREAD

**Discussion by SERichard1974** [home-assistant/frontend#19949] [Automations / scripts / scenes]
Currently when using choice and a numeric value... We have the ability to select an option based on a given range.   But I would like to have the ability for selecting an option based on a given number value not having to specify a range...

Example 1: Using an aqara cube I have a numeric helper where I store the last side it was on... Currently for a few scripts I have to choose based on the number being between 1 & 3 for the value 2, etc.  

Example 2: I have RF controlled fans that I have recently added a broadlink RM4Pro to control them inside the house. These being dumb fans with only input going to them I have to track their current fan speed for a few automations. I store the speed in the values 0-3 (off, low, med, high) in a numeric helper, choice between them. 

In both of these cases Having the ability to just do a value choose rather than having to use a template value expression in the UI vs using YAML do this would be a very big help to alot of users, not just myself.

**Comment by karwosts** [comment]
Have you explored what you can do with dropdown (select) helpers? Those allow you to just define a list of fixed values and select between them easily in the UI.

I don't quite know if that's what you're asking about, but it sounds similar.

**Comment by SERichard1974** [comment]
It’s a kludge workaround… but yes it does work. However simple number choices should also work similar.

**Reply by karwosts** [reply]
Are you trying to use `numeric_state` condition? If you just want to specify an exact value for your number you can just use `state` condition, not numeric_state.
